src/bloqade/squin/rewrite/non_clifford_to_U3.py

```diff
@@ -0,0 +1,105 @@
+from kirin import ir
+from kirin.rewrite import abc as rewrite_abc
+from kirin.dialects import py
+
+from bloqade.squin.gate import stmts as gate_stmts
+
+
+class RewriteNonCliffordToU3(rewrite_abc.RewriteRule):
+    """Rewrite non-Clifford gates to U3 gates.
+
+    This rewrite rule transforms specific non-Clifford single-qubit gates
+    into equivalent U3 gate representations. The following transformations are applied:
+    - T gate (with adjoint attribute) to U3 gate with parameters (0, 0, ±π/4)
+    - Rx gate to U3 gate with parameters (angle, -π/2, π/2)
+    - Ry gate to U3 gate with parameters (angle, 0, 0)
+    - Rz gate is U3 gate with parameters (0, 0, angle)
+
+    This rewrite should be paired with `U3ToClifford` to canonicalize the circuit.
+
+    """
+
+    def rewrite_Statement(self, node: ir.Statement) -> rewrite_abc.RewriteResult:
+        if not isinstance(
```

Comment:
Isn't this `if` redundant with the `self.default` you're getting below?

Reply:
yes, I'll get rid of this.